_inc/client/components/multiple-choice-question/index.jsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { memoize, pick, shuffle, values } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import MultipleChoiceAnswer from './answer';
+import { FormFieldset } from 'components/forms';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const shuffleAnswers = memoize(
+	answers => {
+		const shuffles = shuffle( answers.filter( ( { doNotShuffle } ) => ! doNotShuffle ) );
+		return answers.map( answer => ( answer.doNotShuffle ? answer : shuffles.pop() ) );
+	},
+	answers =>
+		answers.map( answer => values( pick( answer, 'id', 'doNotShuffle' ) ).join( '_' ) ).join( '-' )
```

Comment:
I think either a comment or a function name to explain what the produced key looks like would be nice.

Reply:
fixed, probably over-engineered in the first place 🙂